lib/ansible/modules/cloud/ovirt/ovirt_storage_domains.py

```diff
@@ -501,10 +518,19 @@ def main():
         state = module.params['state']
         control_state(storage_domains_module)
         if state == 'absent':
+            # Pick random available host when host parameter is missing
+            host_param = module.params['host']
+            if not host_param:
+                hosts_service = connection.system_service().hosts_service()
+                hosts_list = hosts_service.list()
+                host_param = next(
+                    (host.name for host in hosts_list if host.status == hoststate.UP),
+                    None
+                )
             ret = storage_domains_module.remove(
                 destroy=module.params['destroy'],
                 format=module.params['format'],
-                host=module.params['host'],
+                host=host_param,
```

Comment:
This must be a proper Python SDK object, so:

```python
host=otypes.Host(
    name=host_param
) if host_param is not None else None,
```

Reply:
Hmm, I got this.
```code
TypeError: The 'host' parameter should be of type 'str', but it is of type 'Host'.
```
So, I 'll keep it `str`

Reply:
Ah, sorry, you are right. It's actually sending:

```
DELETE /ovirt-engine/api/storagedomains/123?host=myhost
```
Not the XML data.